### ISSUE TITLE
React Native Track Player ios Troubleshooting building 100%

### ISSUE THREAD

**Issue by clementortiz134595** [doublesymmetry/react-native-track-player#2402]
**Describe the Bug**
I'm developing a React Native app using react-native-track-player for audio playback. The app works perfectly fine on Android, but I'm facing issues on iOS. Specifically, when building and running the app, it freezes at "Building 100%" and eventually crashes with the following error:
`com.facebook.react.runtime.JavaScript (9): EXC_BAD_ACCESS (code=1, address=0x0)`

![image](https://github.com/user-attachments/assets/b873546b-35be-4ab2-a62d-071925cd9e88)

**Steps To Reproduce**

1. Use react-native-track-player in a React Native app for audio playback.

2. Build and run the app on iOS (simulator or real device).

3. Observe that the app freezes at "Building 100%" and eventually crashes.

**Code To Reproduce**
```
import React, { useEffect } from 'react';
import { View, Button } from 'react-native';
import TrackPlayer,{ Capability } from 'react-native-track-player';

const AudioPlayer: React.FC = () => {
  useEffect(() => {
    const setupPlayer = async () => {
      await TrackPlayer.setupPlayer();

      // Ajout de la piste
      const audioPath = require('../../assets/audio/teddy.mp3');
      await TrackPlayer.add({
        id: 'trackId',
        url: audioPath,
        title: 'Teddy Audio',
        artist: 'Inconnu',
        artwork: 'https://www.example.com/artwork.jpg',
      });

      // Configure les contrôles de notification
      TrackPlayer.updateOptions({
        
        capabilities: [
          Capability.Play,
          Capability.Pause,
          Capability.Stop,
        ],
        compactCapabilities: [
          Capability.Play,
          Capability.Pause,
        ],
        notificationCapabilities: [
          Capability.Play,
          Capability.Pause,
          Capability.Stop,
        ],
      });
    };

    setupPlayer();

    return () => {
      TrackPlayer.stop();
    };
  }, []);

  const playAudio = async () => {
    await TrackPlayer.play();
  };

  const pauseAudio = async () => {
    await TrackPlayer.pause();
  };

  return (
    <View>
      <Button title="Lire l'audio" onPress={playAudio} />
      <Button title="Pause" onPress={pauseAudio} />
    </View>
  );
};

export default AudioPlayer;
```

**Environment Info:**
```
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.0
    wanted: 15.0.0
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.1
    wanted: 0.76.1
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
  ```
  
  
  package.json : 
  
  
  ```
{
  "name": "test",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@react-native-async-storage/async-storage": "^2.0.0",
    "@react-native-clipboard/clipboard": "^1.15.0",
    "@react-native-community/geolocation": "^3.4.0",
    "@react-native-community/slider": "^4.5.5",
    "@react-native-firebase/app": "^21.4.0",
    "@react-native-firebase/auth": "^21.4.0",
    "@react-native-picker/picker": "^2.9.0",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/native-stack": "^6.11.0",
    "@react-navigation/stack": "^6.4.1",
    "@reduxjs/toolkit": "^2.3.0",
    "@rnmapbox/maps": "^10.1.32",
    "package.json": "^2.0.1",
    "react": "18.3.1",
    "react-native": "0.76.1",
    "react-native-fs": "^2.20.0",
    "react-native-gesture-handler": "^2.20.2",
    "react-native-music-control": "^1.4.1",
    "react-native-otp-input": "^1.0.12",
    "react-native-otp-inputs": "^7.4.0",
    "react-native-progress": "^5.0.1",
    "react-native-reanimated": "^3.16.1",
    "react-native-safe-area-context": "^4.12.0",
    "react-native-screens": "^3.35.0",
    "react-native-sound": "^0.11.2",
    "react-native-sound-player": "^0.14.3",
    "react-native-svg": "^15.8.0",
    "react-native-track-player": "^4.1.1",
    "react-native-vector-icons": "^10.2.0",
    "react-native-webview": "^13.12.3",
    "react-redux": "^9.1.2"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/preset-env": "^7.25.3",
    "@babel/runtime": "^7.25.0",
    "@react-native-community/cli": "15.0.0",
    "@react-native-community/cli-platform-android": "15.0.0",
    "@react-native-community/cli-platform-ios": "15.0.0",
    "@react-native/babel-preset": "0.76.1",
    "@react-native/eslint-config": "0.76.1",
    "@react-native/metro-config": "0.76.1",
    "@react-native/typescript-config": "0.76.1",
    "@types/react": "^18.2.6",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.6.3",
    "eslint": "^8.19.0",
    "jest": "^29.6.3",
    "prettier": "2.8.8",
    "react-test-renderer": "18.3.1",
    "typescript": "5.0.4"
  },
  "engines": {
    "node": ">=18"
  }
}

```
**How I can Help**
I have investigated some underlying JS and native code but couldn't pinpoint the exact cause.
I'm open to contributing a Pull Request if I can identify the root cause with assistance.
Any guidance on debugging this issue on the iOS side would be greatly appreciated.

**Comment by mobileteamt** [comment]
Any update on this?

**Comment by clementortiz134595** [comment]
I figured out the issue I was facing on iOS, and it was caused by the new architecture being enabled by default with React Native. Apparently, it creates compatibility issues with certain modules, especially those related to Hermes or other specific configurations.

To resolve this, I had to disable the new architecture on iOS by using the following command:

RCT_NEW_ARCH_ENABLED=0 pod install

After running this command, everything worked perfectly again. If anyone else is having the same issue, give this a try! It really solved the problem for me.🆗
